### ISSUE TITLE
Embedded spec is not flattened and won't work if there are external references.

### ISSUE THREAD

**Issue by kars7e** [go-swagger/go-swagger#1528]
## Problem statement
My swagger spec includes references to external model file, like this:
```yaml
        schema:
          $ref: './models.json#/definitions/Subscription'
```
This works fine when generating the handlers (references are resolved and code properly builds), but I'm receiving a runtime exception when running the server:

```
panic: open ./models.json: no such file or directory
```
when looking at `SwaggerJSON` string, I see reference to `models.json` file instead of expanded model definition.

This doesn't seem to be very useful, as an embedded spec should be self-sufficient (if user is ok with providing external dependency files when running a server, he/she is probably fine with providing the main spec as well).

## Swagger specification
2.0

## Steps to reproduce

Generate the server code based on swagger spec that includes references to an external file.

## Environment
swagger version: 6bc0267
go version: 1.10 
OS:  linux

**Comment by fredbi** [comment]
Could you please post a more complete example? It is difficult to figure out the failure with such reduced information.
Besides, could you please detail what you mean by "running the server"? Is it panicking when starting up? When you post  a specific request? I assume you posted a request to serve the spec, right?

**Comment by kars7e** [comment]
Hey @fredbi, when preparing a reproducible example of my issue, I saw that the newest version of swagger generates main.go which uses `restapi.FlatSwaggerJSON`. I checked, and indeed it's there in my embedded_spec.go as well! I didn't know it's being generated (I use my own main.go in the project).

I might wonder would need a non-flatten embedded spec, but that's a different discussion :-). Anyway, this is a non-issue anymore (and for anybody who finds this Issue: use `restapi.FlatSwaggerJSON` instead of `restapi.SwaggerJSON`).

Thanks and sorry for generating noise :-)
